### PR TITLE
[ENH, MRG] Cast complex evoked to power for visualization

### PR DESCRIPTION
**Pull request by alexrockhill**
Part of https://github.com/mne-tools/mne-python/pull/10920.

### CODE REVIEW

File: mne/viz/misc.py [hunk 1]
```diff
@@ -137,6 +137,10 @@ def plot_cov(cov, info, exclude=(), colorbar=True, proj=False, show_svd=True,
             logger.info('    The projection vectors do not apply to these '
                         'channels.')
 
+    if np.iscomplex(C).any():
```
Comment (agramfort) [inline mne/viz/misc.py:140]: I would rather use https://numpy.org/doc/stable/reference/generated/numpy.iscomplexobj.html

File: mne/viz/tests/test_misc.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 import matplotlib.pyplot as plt
 
 from mne import (read_events, read_cov, read_source_spaces, read_evokeds,
-                 read_dipole, SourceEstimate, pick_events)
+                 read_dipole, SourceEstimate, pick_events, Covariance)
```
Comment (alexrockhill) [inline mne/viz/tests/test_misc.py:18]:
```suggestion
                 read_dipole, SourceEstimate, pick_events)
```

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
I'll let another reviewer merge if happy.

File: mne/time_frequency/tfr.py [hunk 3]
```diff
@@ -2428,6 +2428,10 @@ def _preproc_tfr(data, times, freqs, tmin, tmax, fmin, fmax, mode,
         copy = baseline is not None
     data = rescale(data, times, baseline, mode, copy=copy)
 
+    if np.iscomplexobj(data):
+        logger.info('Converting phase-amplitude to power for visualization')
+        data = (data * data.conj()).real
```
Comment (larsoner) [inline mne/time_frequency/tfr.py:2433]: You are converting power here but for consistency you should leave it as amplitude for consistency (probably)
```suggestion
        data = np.sqrt((data * data.conj()).real)
```

File: mne/time_frequency/tfr.py [hunk 4]
```diff
@@ -2452,7 +2456,7 @@ def _preproc_tfr(data, times, freqs, tmin, tmax, fmin, fmax, mode,
     data = data[:, ifmin:ifmax, itmin:itmax]
 
     if dB:
-        data = 10 * np.log10((data * data.conj()).real)
+        data = 10 * np.log10(data)
```
Comment (larsoner) [inline mne/time_frequency/tfr.py:2459]: If you have complex data this step is unchanged from before (the thing being `log10`ed here is a power, not amplitude). But if you have real data it's now wrong, as the thing being `log10`ed here is an amplitude not a power.

The easiest way to resolve this is with two changes, one here and one above, by keeping everything as amplitude and then using that knowledge here
```suggestion
        data = 20 * np.log10(data)
```

Comment (alexrockhill) [reply mne/time_frequency/tfr.py]: Good catch!!

File: doc/changes/latest.inc [hunk 5]
```diff
@@ -213,6 +213,8 @@ Bugs
 
 - Annotations contained in EDF files are correctly read as UTF-8 according to the EDF specification (:gh:`10963` by `Clemens Brunner`_)
 
+- Fix bug where complex-valued topoplots caused an error, convert to power instead (:gh:`10978` by `Alex Rockhill`_)
```
Comment (agramfort) [inline doc/changes/latest.inc:216]:
```suggestion
- Fix bug where complex-valued topoplots caused an error, convert to amplitude instead (:gh:`10978` by `Alex Rockhill`_)
```

**Review by agramfort — COMMENTED** [review]
just a nitpick. Then +1 for merge

**Pull request merged**